### PR TITLE
Fix an exception to use string interpolation

### PR DESCRIPTION
**Pull request by notpeter**
image

### CODE REVIEW

**Comment by dimrozakis** [comment]
I think that the line in the above traceback is the only one that should be changed. Log statements don't need to use the `%` operator for string formatting. It happens automatically if you pass the strings as extra positional arguments. Doing it like that (with explicitly using `%`) is also more efficient since string formatting will only occur if the configured log level is such that the log entry would actually be printed.

**Comment by notpeter** [comment]
Ah, sounds good. I'll undo and just fix it where needed.

File: bucky/carbon.py [hunk 1]
```diff
@@ -65,19 +65,19 @@ def connect(self):
...
-                log.error("Failed to connect to %s:%s: %s", self.ip, self.port, e)
+                log.error("Failed to connect to %s:%s: %s" % (self.ip, self.port, e))
                 if reconnect_delay > 0:
                     time.sleep(reconnect_delay)
                     if self.backoff_factor:
                         reconnect_delay *= self.backoff_factor
                         if self.backoff_max:
                             reconnect_delay = min(reconnect_delay, self.backoff_max)
-        raise socket.error("Failed to connect to %s:%s after %s attempts", self.ip, self.port, self.max_reconnects)
+        raise socket.error("Failed to connect to %s:%s after %s attempts" % (self.ip, self.port, self.max_reconnects))
```
Comment (dimrozakis) [inline bucky/carbon.py]: As far as I can tell, this is the only line from the diff that should be merged.

Comment (notpeter) [inline bucky/carbon.py]: And now that's the only line that remains. ;)

Comment (dimrozakis) [inline bucky/carbon.py]: @notpeter :thumbsup:

**Comment by trbs** [comment]
Thanks !

**Pull request merged**